qml/MonthComponent.qml

```diff
@@ -338,7 +338,14 @@ Item{
             id: weekDay
             objectName: "weekDay" + index
             width: parent.dayWidth
-            property var day : Qt.locale(Qt.locale().name).standaloneDayName(( Qt.locale().firstDayOfWeek + index), Locale.ShortFormat);
+            property var day : {
+		        if (Qt.locale().name == "pt_PT") {
+                    // Workaround for https://github.com/ubports/ubuntu-touch/issues/510
+                    Qt.locale("pt").standaloneDayName(( Qt.locale().firstDayOfWeek + index), Locale.ShortFormat);
+                } else {
+                    Qt.locale(Qt.locale().name).standaloneDayName(( Qt.locale().firstDayOfWeek + index), Locale.ShortFormat);
+                }
+            }
             text: isYearView ? Qt.locale(Qt.locale().name).standaloneDayName(( Qt.locale().firstDayOfWeek + index), Locale.NarrowFormat) : day
```

Comment:
Rather than keep repeating the calls to `Qt.locale().blah` in multiple places. It might be a bit cleaner to do something like

```qml
readonly property string localeName: Qt.locale().name === "pt_PT" ? "pt" : Qt.locale().name

text: Qt.locale(localeName).standaloneDayName(Qt.locale().firstDayOfWeek + index, isYearView ? Locale.NarrowFormat : Locale.ShortFormat);
```

What do you think?